src/security-archive/src/index.ts

```diff
@@ -0,0 +1,291 @@
+// eslint-disable-next-line import/no-unresolved
+import { DatabaseSync } from 'node:sqlite'
+import { LRUCache } from 'lru-cache'
+import { loadPackageJson } from 'package-json-from-dist'
+import { asDepID, splitDepID, joinDepIDTuple } from '@vltpkg/dep-id'
+import { error } from '@vltpkg/error-cause'
+import { XDG } from '@vltpkg/xdg'
+import { asPackageReportData } from './types.ts'
+import type { DepID } from '@vltpkg/dep-id'
+import type { GraphLike } from '@vltpkg/graph'
+import type { SpecOptions } from '@vltpkg/spec'
+import type {
+  PackageReportData,
+  SecurityArchiveLike,
+  SecurityArchiveRefreshOptions,
+} from './types.ts'
+
+const SOCKET_API_V0_URL = 'https://api.socket.dev/v0/purl?alerts=true'
+const SOCKET_PUBLIC_API_TOKEN =
+  'sktsec_t_--RAN5U4ivauy4w37-6aoKyYPDt5ZbaT5JBVMqiwKo_api'
+
+export const targetSecurityRegisty = 'https://registry.npmjs.org/'
+
+export type JSONItemResponse = {
+  namespace?: `@{string}`
+  name: string
+  version: string
+}
+
+export type LoadedEntry = {
+  ttl: LRUCache.Milliseconds
+  start: LRUCache.Milliseconds
+  value: PackageReportData
+}
+
+export type SecurityArchiveOptions = {
+  /**
+   * An optional value for the path in which to store the sqlite db.
+   */
+  path?: string
+}
+
+/**
+ * Only the public npm registry has information available in the
+ * socket.dev API. This function checks if a given depID is pointing
+ * to the public npm registry and returns `true` if it does.
+ */
+const usesTargetRegistry = (
+  depID: DepID,
+  specOptions: SpecOptions,
+): boolean => {
+  const [nodeType, nodeRegistry] = splitDepID(depID)
+
+  if (nodeType !== 'registry') {
+    return false
+  }
+
+  const reg =
+    nodeRegistry === '' ?
+      specOptions.registry
+    : specOptions.registries?.[nodeRegistry]
+  return reg === targetSecurityRegisty
+}
+
+// Loads the version number to be used in the User-Agent header
+const { version } = loadPackageJson(import.meta.filename) as {
+  version: string
+}
```

Comment:
Need to use the special env var here for builds, e.g:
```suggestion
export const { version } = loadPackageJson(
  import.meta.filename,
  process.env.__VLT_INTERNAL_CLI_PACKAGE_JSON,
) as {
  version: string
}
```